lifecycle-hooks.md

```diff
@@ -0,0 +1,188 @@
+  - [Extension in Action](#extension-in-action)
+    - [Create a new workload Cluster](#create-a-new-workload-cluster)
+    - [Delete the Cluster](#delete-the-cluster)
+    - [(Optional) Block Cluster deletion using Extension Server](#optional-block-cluster-deletion-using-extension-server)
+  - [Clean up](#clean-up)
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Running the Extension
+
+Firstly we will need a simple extension server that needs to run and be able to receive cluster lifecycle events. For this, lets create a deployment in the management cluster that will act as our extension server.
+
+```bash
+kubectl apply -f yamls/extension/test-extension-deployment.yaml
+```
+
+Note that the YAML also contains all necessary resources like Service, Certificates, etc that will make this extension server reachable from within the management cluster.
+
+Let's verify that the extension server is running:
+
+```bash
+kubectl get deployments -n test-extension-system
+```
+
+The output should resemble:
+
+```bash
+NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
+test-extension-manager   1/1     1            1           30s
+```
+
+## Register the Extension
+
+Now that we have a running extension server let's register it with Cluster API. This will inform Cluster API to call the extension server during key workload cluster lifecycle events.
+
+Let's first look at how the registration looks like:
+
+```yaml
+apiVersion: runtime.cluster.x-k8s.io/v1alpha1
+kind: ExtensionConfig
+metadata:
+  name: printer-extension                      // Unique Registration Name
+  annotations:
+    runtime.cluster.x-k8s.io/inject-ca-from-secret: test-extension-system/test-extension-webhook-service-cert
+spec:
+  clientConfig:                                // Information to contact the extension server
+    service:                                   // Reference to a service running in the management cluster
+      name: test-extension-webhook-service
+      namespace: test-extension-system
+```
+
+Let's register the extension using:
+
+```bash
+kubectl apply -f yamls/extension/test-extension-config.yaml
+```
+
+Let us now verify that the extension is properly registered and reachable:
+
+```bash
+kubectl get extensionconfig printer-extension -o jsonpath='{.status.conditions}'
+```
+
+The output should resemble:
+
+```bash
+[{"lastTransitionTime":"2022-10-12T04:55:44Z","status":"True","type":"Discovered"}]
+```
+
+> **Optional**  
+> If you are curious, take a look at `.status.handlers` of the `printer-extension` resource by doing `kubectl get extensionconfig printer-extension -o jsonpath='{.status.handlers}'`. This lists down all the lifecycle events the extension server would like to receive.
+
+## Extension in Action
+
+Now that we have the extension server running and registered with Cluster API let's see it in action. The extension server is configured to receive the following events (non-exhaustive list):
```

Comment:
Could this be 1) an exhaustive list and 2) contain a little bit of detail about when each hook is called on what it could be used for.

Reply:
I am fine with adding a full list a small description of each hook. 

I am not so sure about going into any further details (including examples) as that might be too much information and deviates from the hands-on a little but. What do you think about adding a link to the proposal instead for folks who might be interested in more details?

Reply:
Yeah - +1 to name plus a single sentence on when the hook is called.

Reply:
Seems fixed